closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -198,6 +268,106 @@ public static boolean isProviderAllowed(Class<?> providerClazz) {
         return true;
     }
 
+    /**
+     * Figure out the full profile ID.
+     *
+     * Use the default or user selected profile and attempt to find
+     * an appropriate entry in the java.security properties.
+     *
+     * If a profile cannot be found, or multiple defaults are discovered
+     * for a single profile, an appropriate message is printed and the
+     * system exits.
+     *
+     * @param props the java.security properties
+     */
+    private static void getProfileID(Properties props) {
+        String potentialProfileID = "RestrictedSecurity." + selectedProfile;
+
+        if (selectedProfile.indexOf(".") != -1) {
+            /* The default profile is used, or the user specified the
+             * full <profile.version>.
+             */
+            if (debug != null) {
+                debug.println("Profile specified using full name (i.e., <profile.version>): "
+                        + selectedProfile);
+            }
+            for (Object keyObject : props.keySet()) {
+                // A matching property is found.
```

Comment:
This comment doesn't seem appropriate here.  I don't think it's needed at all. Same in the else case.

Reply:
I restructured the check for the property there and updated the comment to match it.

Reply:
The else case looks good now, but this comment in this place doesn't make sense to me. Maybe if you rewrote it. The next line is checking if the key is a String. The code seem readable enough without any comment here.

Reply:
It's about the if that comes after, so I can put it below that. I can also remove it altogether though.

Reply:
I'm fine with it removed, or if it's closer to the if that it's referring to.

Reply:
Removed altogether.

Reply:
You seem to have removed a comment from the else case, but the comment on line 295 which I was discussing is still there.

Reply:
Your updated comment in the else case was fine, up to you if you want to put it back.

Reply:
It's fine as it is now.

Reply:
Are you going to remove the comment on line 296?

Reply:
Yep. Done.